spec/models/efile_submission_dependent_spec.rb

```diff
@@ -105,4 +105,151 @@
       end
     end
   end
+
+  describe "#schedule_eic_4a?" do
+    let(:dependent) { create :dependent, intake: submission.intake, full_time_student: full_time_student, birth_date: birth_date }
+    let(:efile_submission_dependent) { create :efile_submission_dependent, dependent: dependent, age_during_tax_year: age_during_tax_year }
+
+    before do
+      efile_submission_dependent.efile_submission.tax_return.update(year: DateTime.now.year)
+      efile_submission_dependent.intake.update(primary_birth_date: 50.years.ago)
+      efile_submission_dependent.intake.update(spouse_birth_date: 50.years.ago)
+    end
+
+    context "when the dependent is between 19 and 24 and a full time student" do
+      let(:birth_date) { 22.years.ago }
+      let(:age_during_tax_year) { 22 }
+      let(:full_time_student) { "yes" }
```

Comment:
for some reason i found it impossible to use `EfileSubmissionDependent.create_qualifying_dependent` to create `efile_submission_dependent`—it just refused to create one. as such, we have this awkward duplication of birth date and age during tax year because the method being tested checks A) the dependent's `age_during_tax_year` is between 19 and 24 and B) the dependent is younger than the filer(s) by comparing their birth dates.

we started out comparing `age_during_tax_year` to a calculation of the filers' age during tax year, but i got confused by that because i was worried that the calculation of a filer's age during tax year could be off by one (since one is 2 ages, respectively, during any given tax year). would love to talk this through with someone!

Reply:
💀